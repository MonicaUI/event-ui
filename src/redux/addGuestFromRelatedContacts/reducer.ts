const TOGGLE_RELATED_CONTACTS_VIEW = 'event-guestside-site/addGuestFromRelatedContacts/LIST_RELATED_CONTACTS';
const LOAD_RELATED_CONTACTS_FOR_INVITEE =
  'event-guestside-site/addGuestFromRelatedContacts/LOAD_RELATED_CONTACTS_FOR_INVITEE';
const LOAD_RELATED_CONTACTS_SEARCH_RESULTS =
  'event-guestside-site/addGuestFromRelatedContacts/LOAD_RELATED_CONTACTS_SEARCH_RESULTS';

export function setRelatedContacts(contactId: $TSFixMe, relatedContacts: $TSFixMe): $TSFixMe {
  return { type: LOAD_RELATED_CONTACTS_FOR_INVITEE, payload: { contactId, relatedContacts } };
}

export function toggleRelatedContacts(): $TSFixMe {
  return { type: TOGGLE_RELATED_CONTACTS_VIEW };
}

export function setRelatedContactsSearch(
  contactId: $TSFixMe,
  searchCriteria: $TSFixMe,
  searchResults: $TSFixMe
): $TSFixMe {
  return { type: LOAD_RELATED_CONTACTS_SEARCH_RESULTS, payload: { contactId, searchCriteria, searchResults } };
}

export default function addGuestFromRelatedContactsReducer(state = {}, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case LOAD_RELATED_CONTACTS_FOR_INVITEE: {
      return {
        ...state,
        [action.payload.contactId]: {
          ...state[action.payload.contactId],
          relatedContacts: action.payload.relatedContacts
        }
      };
    }
    case TOGGLE_RELATED_CONTACTS_VIEW: {
      return {
        ...state,
        showRelatedContactsView: !(state as $TSFixMe).showRelatedContactsView
      };
    }
    case LOAD_RELATED_CONTACTS_SEARCH_RESULTS: {
      return {
        ...state,
        [action.payload.contactId]: {
          ...state[action.payload.contactId],
          relatedContactsSearchCriteria: action.payload.searchCriteria,
          relatedContactsSearchResults: action.payload.searchResults
        }
      };
    }
    default: {
      return state;
    }
  }
}
