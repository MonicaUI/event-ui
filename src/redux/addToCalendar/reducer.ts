import { CALENDAR_EVENT_ID } from '../../utils/addToCalendarUtils';

export const LOAD_ADD_TO_CALENDAR_PROVIDERS = 'event-guestside-site/calendar/LOAD_CALENDAR_PROVIDERS';
export const LOAD_EVENT_CALENDAR = 'event-guestside-site/calendar/LOAD_EVENT_CALENDAR';
export const LOAD_SESSION_CALENDAR = 'event-guestside-site/calendar/LOAD_SESSION_CALENDAR';

export function setAddToCalendarProviders(calendarProviders: $TSFixMe): $TSFixMe {
  return {
    type: LOAD_ADD_TO_CALENDAR_PROVIDERS,
    payload: { calendarProviders }
  };
}

export function setCalendarUrl(calendarType: $TSFixMe, calendarUrl: $TSFixMe, entityId = CALENDAR_EVENT_ID): $TSFixMe {
  if (entityId === CALENDAR_EVENT_ID) {
    return {
      type: LOAD_EVENT_CALENDAR,
      payload: { calendarType, calendarUrl }
    };
  }

  return {
    type: LOAD_SESSION_CALENDAR,
    payload: { calendarType, calendarUrl, entityId }
  };
}

export default function addToCalendarReducer(state = {}, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case LOAD_ADD_TO_CALENDAR_PROVIDERS: {
      return {
        ...state,
        ...action.payload.calendarProviders
      };
    }
    case LOAD_EVENT_CALENDAR: {
      return {
        ...state,
        [action.payload.calendarType]: {
          ...state[action.payload.calendarType],
          [CALENDAR_EVENT_ID]: action.payload.calendarUrl
        }
      };
    }
    case LOAD_SESSION_CALENDAR: {
      return {
        ...state,
        [action.payload.calendarType]: {
          ...state[action.payload.calendarType],
          [action.payload.entityId]: action.payload.calendarUrl
        }
      };
    }
    default: {
      return state;
    }
  }
}
