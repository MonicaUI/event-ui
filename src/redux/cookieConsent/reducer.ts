// Reducer to update state for cookie Consent
import { DENY } from '../../utils/CookieConstants';

export const SET_COOKIE_CONSENT = 'event-guestside-site/cookieConsent/SET_COOKIE_CONSENT';

export default function cookieConsentReducer(state = { status: DENY }, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case SET_COOKIE_CONSENT:
      return {
        ...state,
        status: action.payload.cookieStatus
      };
    default:
      return state;
  }
}
