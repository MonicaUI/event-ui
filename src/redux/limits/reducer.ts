const SET_ACCOUNT_LIMITS = 'event-guestside-site/limits/SET_ACCOUNT_LIMITS';

export const setAccountLimits = (accountLimits: $TSFixMe): $TSFixMe => {
  return {
    type: SET_ACCOUNT_LIMITS,
    payload: {
      accountLimits
    }
  };
};

export default function reducer(state: $TSFixMe, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case SET_ACCOUNT_LIMITS: {
      return action.payload.accountLimits;
    }
    default: {
      return state;
    }
  }
}
