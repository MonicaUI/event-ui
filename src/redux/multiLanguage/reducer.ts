import { SET_LOCALE } from 'nucleus-widgets/redux/modules/text';
import { LOAD_LANGUAGE } from './actions';

export default function reducer(state: $TSFixMe, action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case SET_LOCALE:
      return {
        ...state,
        locale: action.payload.locale
      };
    case LOAD_LANGUAGE:
      return {
        ...state,
        loadedLanguages: [...state.loadedLanguages, action.payload]
      };
    default:
      return state;
  }
}
