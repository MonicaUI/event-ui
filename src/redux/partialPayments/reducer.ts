import { UPDATE_PAYMENT_AMOUNT, UPDATE_SELECTED_PAYMENT_OPTION } from './actions';
import { PAYMENT_AMOUNT_OPTION } from 'event-widgets/utils/paymentConstant';
import { START_MODIFICATION_SUCCESS } from '../registrationForm/regCart/actionTypes';
import { GET_ORDERS } from '../actionTypes';

// Reducer to update state for Partial Payments
export default function reducer(state: $TSFixMe, action: $TSFixMe): $TSFixMe {
  const defaultPartialPayment = { paymentAmountOption: null, paymentAmount: null };
  switch (action.type) {
    case UPDATE_SELECTED_PAYMENT_OPTION: {
      const FULL_PAYMENT_VALUE = PAYMENT_AMOUNT_OPTION.FULL_PAYMENT.value;
      const paymentOption =
        action.payload === FULL_PAYMENT_VALUE
          ? PAYMENT_AMOUNT_OPTION.FULL_PAYMENT
          : PAYMENT_AMOUNT_OPTION.PARTIAL_PAYMENT;
      return {
        ...state,
        paymentAmountOption: paymentOption
      };
    }
    case UPDATE_PAYMENT_AMOUNT:
      return {
        ...state,
        paymentAmount: action.payload
      };
    case START_MODIFICATION_SUCCESS:
    case GET_ORDERS:
      return defaultPartialPayment;
    default:
      return state;
  }
}
