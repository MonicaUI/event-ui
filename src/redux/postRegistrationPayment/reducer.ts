import {
  RESET_SUBMIT_WEB_PAYMENT,
  SUBMIT_WEB_PAYMENT,
  STORE_WEBPAYMENT_DATA,
  FINALIZE_POSTREG_CHECKOUT_PENDING,
  FINALIZE_POSTREG_CHECKOUT_COMPLETE
} from './actionTypes';
import { PAYMENT_TYPE } from 'event-widgets/constants/PaymentType';
import { addQueryParams } from '../../utils/queryUtils';
import { PAYMENT_AMOUNT_OPTION } from 'event-widgets/utils/paymentConstant';

export default function reducer(state = '', action: $TSFixMe): $TSFixMe {
  switch (action.type) {
    case SUBMIT_WEB_PAYMENT: {
      return {
        // @ts-expect-error ts-migrate(2698) FIXME: Spread types may only be created from object types... Remove this comment to see the full error message
        ...state,
        submitWebPayment: true
      };
    }
    case RESET_SUBMIT_WEB_PAYMENT:
      return {
        // @ts-expect-error ts-migrate(2698) FIXME: Spread types may only be created from object types... Remove this comment to see the full error message
        ...state,
        submitWebPayment: false
      };
    case STORE_WEBPAYMENT_DATA:
      return {
        // @ts-expect-error ts-migrate(2698) FIXME: Spread types may only be created from object types... Remove this comment to see the full error message
        ...state,
        webPaymentData: action.payload.webPaymentData
      };
    case FINALIZE_POSTREG_CHECKOUT_PENDING:
      return {
        // @ts-expect-error ts-migrate(2698) FIXME: Spread types may only be created from object types... Remove this comment to see the full error message
        ...state,
        isCheckingOut: true
      };
    case FINALIZE_POSTREG_CHECKOUT_COMPLETE:
      return {
        // @ts-expect-error ts-migrate(2698) FIXME: Spread types may only be created from object types... Remove this comment to see the full error message
        ...state,
        isCheckingOut: false
      };
    default:
      return {
        // @ts-expect-error ts-migrate(2698) FIXME: Spread types may only be created from object types... Remove this comment to see the full error message
        ...state,
        submitWebPayment: false
      };
  }
}

function getPaymentDistributions(orders) {
  const paymentDistribution = new Map();
  orders.forEach(order => {
    if (order.orderType === 'OfflineCharge') {
      order.orderItems.forEach(item => {
        if (item.amountDue > 0) {
          if (paymentDistribution.get(order.attendeeId)) {
            paymentDistribution.get(order.attendeeId).orders.push({
              orderId: order.orderId,
              orderDetailId: item.itemId,
              amountIntended: item.amountDue
            });
          } else {
            paymentDistribution.set(order.attendeeId, {
              attendeeId: order.attendeeId,
              orders: [
                {
                  orderId: order.orderId,
                  orderDetailId: item.itemId,
                  amountIntended: item.amountDue
                }
              ]
            });
          }
        }
      });
    }
  });
  return Array.from(paymentDistribution.values());
}

export function getTotals(orders: $TSFixMe): $TSFixMe {
  let subtotal = 0;
  let total = 0;
  if (orders) {
    orders.forEach(order => {
      if (order.orderType === 'OfflineCharge') {
        order.orderItems.forEach(item => {
          total = +(total + item.amountDue).toFixed(12);
          if (
            item.productType !== 'Tax' &&
            item.productType !== 'RegistrationTypeServiceFee' &&
            item.productType !== 'PaymentTypeServiceFee'
          ) {
            subtotal = +(subtotal + item.amountDue).toFixed(12);
          }
        });
      }
    });
  }
  return { subtotal, total };
}

export function getPostRegPaymentPricingInfo(
  orders: $TSFixMe,
  webPaymentData: $TSFixMe,
  accessToken: $TSFixMe,
  partialPaymentSetting?: $TSFixMe
): $TSFixMe {
  const token = accessToken.split(' ');
  const totalAmount =
    partialPaymentSetting?.paymentAmountOption === PAYMENT_AMOUNT_OPTION.PARTIAL_PAYMENT
      ? partialPaymentSetting.paymentAmount
      : getTotals(orders).total;
  return {
    postRegPaymentData: {
      totalAmount,
      cardType: webPaymentData.cardType,
      paymentDistribution: getPaymentDistributions(orders)
    },
    continueCheckout: true,
    checkoutInfo: {
      paymentMethodType: webPaymentData.cardType,
      paymentType: PAYMENT_TYPE.ONLINE,
      paymentAmount: totalAmount,
      paymentDetail: {
        amount: totalAmount,
        paymentMethodMode: 'UseNewPaymentMethod',
        browserLandingURL: addQueryParams(window.location.href, { fromRedirect: 'true' }).replace(/(^\w+:|^)\/\//, ''),
        contextId: webPaymentData.contextId,
        coreBearerToken: token[1]
      }
    }
  };
}

/**
 * This method is called only for Service Fees with Post Reg Payments
 * to pass checkout info object when startRegCartCheckout is called for
 * continuing checkout
 */
export function getCheckoutInfoForPostServiceFees(
  webPaymentData: $TSFixMe,
  regCartPricing: $TSFixMe,
  accessToken: $TSFixMe
): $TSFixMe {
  const token = accessToken.split(' ');
  const totalAmount = regCartPricing.netFeeAmountChargeWithPaymentAmountServiceFee;
  return {
    paymentMethodType: webPaymentData.cardType,
    paymentType: PAYMENT_TYPE.ONLINE,
    paymentAmount: totalAmount,
    paymentDetail: {
      amount: totalAmount,
      paymentMethodMode: 'UseNewPaymentMethod',
      browserLandingURL: addQueryParams(window.location.href, { fromRedirect: 'true' }).replace(/(^\w+:|^)\/\//, ''),
      contextId: webPaymentData.contextId,
      coreBearerToken: token[1]
    }
  };
}
