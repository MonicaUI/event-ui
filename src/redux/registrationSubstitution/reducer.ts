export const SUBSTITUTE_REGISTRANT_SUCCESS = 'event-guestside-site/substituteRegistrant/SUBSTITUTE_REGISTRANT_SUCCESS';
export const SUBSTITUTE_REGISTRANT_ERROR = 'event-guestside-site/substituteRegistrant/SUBSTITUTE_REGISTRANT_ERROR';
export const RESET_SUBSTITUTE_REGISTRANT = 'event-guestside-site/substituteRegistrant/RESET_SUBSTITUTE_REGISTRANT';
export const SET_FIELD = 'event-guestside-site/substituteRegistrant/SET_FIELD';
export const SET_VALIDATION = 'event-guestside-site/substituteRegistrant/SET_VALIDATION';
export const AUTO_FOCUS_ON = 'event-guestside-site/substituteRegistrant/AUTO_FOCUS_ON';
export const REQUEST_CONFIRMATION = 'event-guestside-site/substituteRegistrant/REQUEST_CONFIRMATION';
export const DENIED_CONFIRMATION = 'event-guestside-site/substituteRegistrant/DENIED_CONFIRMATION';
export const ACCEPTED_CONFIRMATION = 'event-guestside-site/substituteRegistrant/ACCEPTED_CONFIRMATION';
export const SET_SUBSTITUTION_CART = 'event-guestside-site/substituteRegistrant/SET_SUBSTITUTION_CART';
export const SHOW_CONCURRENT_ACTION_POPUP = 'event-guestside-site/substituteRegistrant/SHOW_CONCURRENT_ACTION_POPUP';
export const SHOW_CART_ABORTED_MESSAGE = 'event-guestside-site/substituteRegistrant/SHOW_CART_ABORTED_MESSAGE';
export const SHOW_SUBSTITUTION_FORM = 'event-guestside-site/substituteRegistrant/SHOW_SUBSTITUTION_FORM';
export const DISABLE_SUBMIT_BUTTON = 'event-guestside-site/substituteRegistrant/DISABLE_SUBMIT_BUTTON';

export function showConcurrentActionPopupMessage(substitutionCart: $TSFixMe): $TSFixMe {
  return { type: SHOW_CONCURRENT_ACTION_POPUP, payload: { substitutionCart } };
}

export function hideConcurrentActionPopupMessage(): $TSFixMe {
  return { type: SHOW_SUBSTITUTION_FORM };
}

export function showCartAbortedMessage(): $TSFixMe {
  return { type: SHOW_CART_ABORTED_MESSAGE };
}

export function substituteRegistrationSuccess(): $TSFixMe {
  return { type: SUBSTITUTE_REGISTRANT_SUCCESS };
}

export function substituteRegistrationError(): $TSFixMe {
  return { type: SUBSTITUTE_REGISTRANT_ERROR };
}

export function resetSubstituteRegistration(): $TSFixMe {
  return { type: RESET_SUBSTITUTE_REGISTRANT };
}

export function setShowConfirmationMessage(substitutionCart: $TSFixMe): $TSFixMe {
  return { type: REQUEST_CONFIRMATION, payload: { substitutionCart } };
}

export function setConfirmationDenied(): $TSFixMe {
  return { type: DENIED_CONFIRMATION };
}

export function setConfirmationAccepted(): $TSFixMe {
  return { type: ACCEPTED_CONFIRMATION };
}

export const setSubstituteRegistrantFields = (fieldName: $TSFixMe, fieldValue: $TSFixMe): $TSFixMe => {
  return { type: SET_FIELD, payload: { fieldName, fieldValue: fieldValue.trim() } };
};

export function turnOnAutoFocus(): $TSFixMe {
  return { type: AUTO_FOCUS_ON };
}

export function disableSubmitButton(): $TSFixMe {
  return { type: DISABLE_SUBMIT_BUTTON };
}

export function setValidationMessage(validation: $TSFixMe, shouldSubmitButtonShouldBeDisabled?: $TSFixMe): $TSFixMe {
  return { type: SET_VALIDATION, payload: { validation, shouldSubmitButtonShouldBeDisabled } };
}

export function setSubstitutionCart(substitutionCart: $TSFixMe): $TSFixMe {
  return {
    type: SET_SUBSTITUTION_CART,
    payload: { substitutionCart }
  };
}

const shouldSubmitButtonBeEnabled = (state, dupMatchKeyType, action) => {
  if (state.disableSubmitButton) {
    if (dupMatchKeyType === 'EMAIL_ONLY') {
      return (
        action.payload.fieldName === 'emailAddress' && state.substitutionForm.emailAddress !== action.payload.fieldValue
      );
    }
    return state.substitutionForm?.[action.payload.fieldName] !== action.payload.fieldValue;
  }
  return true;
};

export const initialState = {
  substituteRegistrationSuccess: false,
  substituteRegistrationError: false,
  showConfirmationMessage: false,
  hasConfirmed: false,
  autoFocus: false,
  showConcurrentActionMessage: false,
  cartAborted: false,
  substitutionForm: {
    firstName: '',
    lastName: '',
    emailAddress: ''
  },
  validationList: null,
  substitutionCart: null,
  originalSubstitutionCart: null,
  disableSubmitButton: false
};

export default function reducer(state = initialState, action = {}, dupMatchKeyType = 'EMAIL_ONLY'): $TSFixMe {
  switch ((action as $TSFixMe).type) {
    case SUBSTITUTE_REGISTRANT_SUCCESS:
      return {
        ...state,
        originalSubstitutionCart: null,
        substituteRegistrationSuccess: true,
        substituteRegistrationError: false,
        substitutionCart: {
          ...state.substitutionCart,
          status: 'COMPLETED'
        }
      };

    case SUBSTITUTE_REGISTRANT_ERROR:
      return {
        ...state,
        substituteRegistrationError: true,
        substituteRegistrationSuccess: false,
        showConcurrentActionMessage: false,
        cartAborted: false,
        substitutionCart: {
          ...state.substitutionCart,
          status: 'FAILED'
        }
      };

    case RESET_SUBSTITUTE_REGISTRANT:
      return initialState;

    case SET_FIELD:
      return {
        ...state,
        disableSubmitButton: !shouldSubmitButtonBeEnabled(state, dupMatchKeyType, action),
        substitutionForm: {
          ...state.substitutionForm,
          [(action as $TSFixMe).payload.fieldName]: (action as $TSFixMe).payload.fieldValue
        },
        validationList: null
      };

    case AUTO_FOCUS_ON:
      return {
        ...state,
        autoFocus: true
      };

    case REQUEST_CONFIRMATION:
      return {
        ...state,
        showConfirmationMessage: true,
        validationList: null,
        substitutionCart: (action as $TSFixMe).payload.substitutionCart
      };

    case DENIED_CONFIRMATION:
      return {
        ...state,
        showConfirmationMessage: false
      };

    case ACCEPTED_CONFIRMATION:
      return {
        ...state,
        showConfirmationMessage: false,
        hasConfirmed: true
      };

    case SET_VALIDATION:
      return {
        ...state,
        validationList: (action as $TSFixMe).payload.validation,
        disableSubmitButton: (action as $TSFixMe).payload.shouldSubmitButtonShouldBeDisabled
      };

    case SET_SUBSTITUTION_CART:
      return {
        ...state,
        substitutionCart: (action as $TSFixMe).payload.substitutionCart,
        substitutionForm: {
          firstName: (action as $TSFixMe).payload.substitutionCart.substituentInformation.firstName,
          lastName: (action as $TSFixMe).payload.substitutionCart.substituentInformation.lastName,
          emailAddress: (action as $TSFixMe).payload.substitutionCart.substituentInformation.emailAddress
        }
      };

    case SHOW_CONCURRENT_ACTION_POPUP: {
      return {
        ...state,
        showConcurrentActionMessage: true,
        originalSubstitutionCart: (action as $TSFixMe).payload.substitutionCart
      };
    }

    case SHOW_CART_ABORTED_MESSAGE: {
      return {
        ...state,
        cartAborted: true
      };
    }

    case SHOW_SUBSTITUTION_FORM: {
      return {
        ...state,
        showConcurrentActionMessage: false,
        originalSubstitutionCart: null
      };
    }

    case DISABLE_SUBMIT_BUTTON: {
      return {
        ...state,
        disableSubmitButton: true
      };
    }

    default:
      return state;
  }
}
