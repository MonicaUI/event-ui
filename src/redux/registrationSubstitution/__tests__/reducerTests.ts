/* global */
import reducer, {
  disableSubmitButton,
  initialState,
  setSubstitutionCart,
  showCartAbortedMessage,
  setSubstituteRegistrantFields,
  resetSubstituteRegistration,
  setValidationMessage,
  setConfirmationAccepted,
  setConfirmationDenied,
  setShowConfirmationMessage,
  turnOnAutoFocus,
  substituteRegistrationError,
  substituteRegistrationSuccess,
  hideConcurrentActionPopupMessage,
  showConcurrentActionPopupMessage
} from '../reducer';

const FIRST_LAST_EMAIL = 'FIRST_LAST_EMAIL';

describe('Test Reducer :', () => {
  const state = initialState;
  let newState = state;
  test('Verifying initial state.', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });
  test('Verifying turnOnAutoFocus', () => {
    newState = reducer(newState, turnOnAutoFocus());
    expect(newState.autoFocus).toBeTruthy();
  });
  test('verifying setSubstituteRegistrantFields and setSubstitutionCart', () => {
    const cart = {
      substituentInformation: {
        firstName: 'firstName1',
        lastName: 'lastName1',
        emailAddress: 'emailAddress1'
      }
    };
    newState = reducer(newState, setSubstituteRegistrantFields('firstName', 'firstName'));
    newState = reducer(newState, setSubstituteRegistrantFields('lastName', 'lastName'));
    newState = reducer(newState, setSubstituteRegistrantFields('emailAddress', 'emailAddress'));
    expect(newState.substitutionForm.firstName).toEqual('firstName');
    expect(newState.substitutionForm.lastName).toEqual('lastName');
    expect(newState.substitutionForm.emailAddress).toEqual('emailAddress');

    newState = reducer(newState, setSubstitutionCart(cart));
    expect(newState.substitutionForm.firstName).toEqual(cart.substituentInformation.firstName);
    expect(newState.substitutionForm.lastName).toEqual(cart.substituentInformation.lastName);
    expect(newState.substitutionForm.emailAddress).toEqual(cart.substituentInformation.emailAddress);
    expect(newState.substitutionCart).toEqual(cart);

    newState = reducer(newState, disableSubmitButton());
    newState = reducer(newState, setSubstituteRegistrantFields('firstName', 'firstName'));
    expect(newState.disableSubmitButton).toBeTruthy();
    newState = reducer(newState, setSubstituteRegistrantFields('emailAddress', 'emailAddress1'));
    expect(newState.disableSubmitButton).toBeTruthy();
    newState = reducer(newState, setSubstituteRegistrantFields('emailAddress', 'emailAddress2'));
    expect(newState.disableSubmitButton).toBeFalsy();

    newState = reducer(newState, disableSubmitButton());
    newState = reducer(newState, setSubstituteRegistrantFields('firstName', 'firstName'), FIRST_LAST_EMAIL);
    expect(newState.disableSubmitButton).toBeTruthy();
    newState = reducer(newState, setSubstituteRegistrantFields('lastName', 'firstName1'), FIRST_LAST_EMAIL);
    expect(newState.disableSubmitButton).toBeFalsy();
  });
  test('verifying setValidationMessage when validated', () => {
    newState = reducer(newState, setValidationMessage(['Validation']));
    expect(newState.validationList).toStrictEqual(['Validation']);
  });
  test('verifying setShowConfirmationMessage', () => {
    newState = reducer(newState, setShowConfirmationMessage({ substitutionCartId: 'substitutionCartId' }));
    expect(newState.validationList).toBe(null);
    expect(newState.substitutionCart).toStrictEqual({ substitutionCartId: 'substitutionCartId' });
    expect(newState.showConfirmationMessage).toBeTruthy();
  });
  test('verifying setConfirmationDenied', () => {
    newState = reducer(newState, setConfirmationDenied());
    expect(newState.validationList).toBe(null);
    expect(newState.showConfirmationMessage).toBeFalsy();
  });
  test('verifying setConfirmationAccepted', () => {
    newState = reducer(newState, setConfirmationAccepted());
    expect(newState.hasConfirmed).toBeTruthy();
  });
  test('verifying substituteRegistrationSuccess', () => {
    newState = reducer(newState, substituteRegistrationSuccess());
    expect(newState.substituteRegistrationSuccess).toBeTruthy();
    expect(newState.originalSubstitutionCart).toEqual(null);
  });
  test('verifying substituteRegistrationError', () => {
    newState = reducer(newState, substituteRegistrationError());
    expect(newState.substituteRegistrationError).toBeTruthy();
    expect(newState.showConcurrentActionMessage).toBeFalsy();
    expect(newState.cartAborted).toBeFalsy();
  });
  test('verifying resetSubstituteRegistration', () => {
    newState = reducer(newState, resetSubstituteRegistration());
    expect(newState).toEqual(state);
  });
  test('verifying cartAborted', () => {
    newState = reducer(newState, showCartAbortedMessage());
    expect(newState.cartAborted).toBeTruthy();
  });
  test('verifying showConcurrentActionMessage', () => {
    const cart = {
      substituentInformation: {
        firstName: 'firstName1',
        lastName: 'lastName1',
        emailAddress: 'emailAddress1'
      }
    };
    newState = reducer(newState, showConcurrentActionPopupMessage(cart));
    expect(newState.showConcurrentActionMessage).toBeTruthy();
    expect(newState.originalSubstitutionCart).toEqual(cart);
  });
  test('verifying showSubstitutionForm', () => {
    newState = reducer(newState, hideConcurrentActionPopupMessage());
    expect(newState.showConcurrentActionMessage).toBeFalsy();
    expect(newState.originalSubstitutionCart).toEqual(null);
  });
});
