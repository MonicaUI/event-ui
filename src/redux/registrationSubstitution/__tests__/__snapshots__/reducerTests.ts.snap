// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Reducer : Verifying initial state. 1`] = `
Object {
  "autoFocus": false,
  "cartAborted": false,
  "disableSubmitButton": false,
  "hasConfirmed": false,
  "originalSubstitutionCart": null,
  "showConcurrentActionMessage": false,
  "showConfirmationMessage": false,
  "substituteRegistrationError": false,
  "substituteRegistrationSuccess": false,
  "substitutionCart": null,
  "substitutionForm": Object {
    "emailAddress": "",
    "firstName": "",
    "lastName": "",
  },
  "validationList": null,
}
`;
